cockroachcloud/configure-cloud-org-sso.md

```diff
@@ -0,0 +1,213 @@
+1. Click **Save**.
+
+### Autoprovisioning
+
+By default, autoprovisioning is disabled. In order to log in to {{ site.data.products.db }} using SSO:
+
+- A user's email address must exist in the SSO IdP.
+- An organizational admin user must have already invited the user to the {{ site.data.products.db }} organization.
+
+Autoprovisioning allows members to access your organization without an invitation. Users are assigned the [Developer role](/docs/cockroachcloud/console-access-management.html#developer) by default. If you enable autoprovisioning, Cockroach Labs recommends that you also limit the [allowed email domains](#allowed-email-domains) for the authentication method.
+
+To enable autoprovisioning for an SSO authentication method:
+
+1. Log in to [{{ site.data.products.db }} Console](https://cockroachlabs.cloud) as an organizational admin.
+1. Go to **Organization** > **Authentication**.
+1. Click the name of an authentication method.
+1. Click **Advanced Settings**.
+1. Next to **Autoprovisioning**, click **Enable**.
+
+## Add a custom authentication method
+
+You can add a custom authentication method to connect to any IdP that supports [Security Access Markup Language (SAML)](https://en.wikipedia.org/wiki/Security_Assertion_Markup_Language) or [OpenID Connect (OIDC)](https://openid.net/connect/).
+
+### OIDC
+
+To configure a custom OIDC authentication method:
+
+1. Log in to your IdP and gather the following information, which you will use to configure {{ site.data.products.db }} SSO:<ul><li>Issuer URL</li><li>Client ID</li><li>Client secret</li><li>Callback URL</li></ul>
+1. In a separate browser, log in to [{{ site.data.products.db }} Console](https://cockroachlabs.cloud) as an organizational admin.
+1. Go to **Organization** > **Authentication**.
+1. Next to **Authentication Methods**, click **Add**.
+1. Set **Configuration** to **OIDC (OpenID Connect)**.
+1. Set **Provider Name** to a display name for the connection.
+1. Click **Next**.
+1. The **Provider Details** page displays.
+1. To edit the connection, click **Edit**.
+1. Paste the values from your IdP for the **Issuer URL**, **Client ID**, **Client Secret**, and **Callback URL**.
+1. Click **Save**.
+1. The authentication method has been added but is disabled. To enable it, toggle **Enable**.
+1. Click **Test**. If errors are shown, edit the configuration to fix the problems and try again.
+1. Optionally, [configure advanced settings](#configure-advanced-settings) for the new authentication method.
+
+### SAML
+
+To configure a custom SAML authentication method:
+
+1. Log in to your IdP and gather the following information, which you will use to configure {{ site.data.products.db }} SSO:<ul><li>Sign-in URL</li><li>Signing certificate</li></ul>
```

Comment:
Just went through this process to set up SSO for @MaterializeInc! It was mostly smooth, with one hiccup: I had to configure our IdP (Okta) to send along two SAML assertions.

  * `name` (unspecified name format), set to `${user.firstName} ${user.lastName}`
  * `email` (unspecified name format), set to `user.email`

It'd be great to get these requirements documented more formally, because I'm not totally sure I did it right. I figured the above out via trial and error—I tried a few different assertions and retried the "test connection" until I saw what I wanted.

Reply:
Thanks for this, see line 189. :)